YDotNet/Document/Transactions/Transaction.cs

```diff
@@ -32,7 +33,12 @@ internal Transaction(nint handle, Doc doc)
         doc.NotifyTransactionStarted();
     }
 
-    /// <inheritdoc/>
+    /// <summary>
+    ///     Gets a value indicating whether the transaction is writeable.
+    /// </summary>
+    public bool Writeable => TransactionChannel.Writeable(Handle) == 1;
```

Comment:
In general we already have this information, right? So we could just pass it in over the ctor.